File: src/BaseHmiTypes/Screens/Base/HmiDynamicSvg.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiDynamicSvg : HmiCustomWidgetContainer
{
    public HmiDynamicSvg()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiDynamicSvg;
    }

    public HmiProperty<HmiDynamicSvgType>? SvgType { get; set; }

    public HmiProperty<HmiImageSource>? Image { get; set; }

    public HmiProperty<HmiColor>? TransparentColor { get; set; }

    public HmiProperty<bool>? UseTransparentColor { get; set; }

    public HmiProperty<bool>? UseDesignColorSchema { get; set; }

    public IList<HmiDynamicSvgProperty> Properties { get; } = new List<HmiDynamicSvgProperty>();
}

File: typescript/base-hmi-types/src/screens/base/HmiDynamicSvg.ts
import { HmiCustomWidgetContainer } from "./HmiCustomWidgetContainer.js";
import { HmiColor } from "./HmiColor.js";
import { HmiDynamicSvgProperty } from "./HmiDynamicSvgProperty.js";
import { HmiDynamicSvgType } from "./HmiDynamicSvgType.js";
import { HmiImageSource } from "./HmiImageSource.js";
import { HmiProperty } from "./HmiProperty.js";
import { HmiObjectType } from "./HmiObjectType.js";

export class HmiDynamicSvg extends HmiCustomWidgetContainer {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiDynamicSvg;
  }

  svgType?: HmiProperty<HmiDynamicSvgType>;
  image?: HmiProperty<HmiImageSource>;
  transparentColor?: HmiProperty<HmiColor>;
  useTransparentColor?: HmiProperty<boolean>;
  useDesignColorSchema?: HmiProperty<boolean>;
  readonly properties: HmiDynamicSvgProperty[] = [];
}
